packages/SwingSet/src/kernel/collectionManager.js

```diff
@@ -0,0 +1,937 @@
+    const collection = summonCollectionInternal(false, 'GC', subid, kindName);
+
+    const doMoreGC = collection.clearInternal();
+    let priorKey = '';
+    const keyPrefix = prefixc(subid, '|');
+    while (priorKey !== undefined) {
+      [priorKey] = syscall.vatstoreGetAfter(priorKey, keyPrefix);
+      if (!priorKey) {
+        break;
+      }
+      syscall.vatstoreDelete(priorKey);
+    }
+    return doMoreGC;
+  }
+
+  let nextCollectionID = 1;
+
+  function makeCollection(label, kindName, keySchema, valueSchema) {
+    assert.typeof(label, 'string');
+    assert(storeKindInfo[kindName]);
+    assertKeyPattern(keySchema);
+    const schemata = [keySchema];
+    if (valueSchema) {
+      assertPattern(valueSchema);
+      schemata.push(valueSchema);
+    }
+    const collectionID = nextCollectionID;
+    nextCollectionID += 1;
+    const kindID = obtainStoreKindID(kindName);
+    const vobjID = `o+${kindID}/${collectionID}`;
+
+    syscall.vatstoreSet(prefixc(collectionID, '|nextOrdinal'), '1');
+    syscall.vatstoreSet(
+      prefixc(collectionID, '|schemata'),
+      JSON.stringify(serialize(harden(schemata))),
+    );
+    syscall.vatstoreSet(prefixc(collectionID, '|label'), label);
+
+    return [
+      vobjID,
+      summonCollection(
+        true,
+        label,
+        collectionID,
+        kindName,
+        keySchema,
+        valueSchema,
+      ),
+    ];
+  }
+
+  function collectionToMapStore(collection) {
+    return Far('mapStore', collection);
+  }
+
+  function collectionToWeakMapStore(collection) {
+    return Far('weakMapStore', collection);
+  }
+
+  function collectionToSetStore(collection) {
+    const {
+      has,
+      init,
+      delete: del,
+      keys,
+      size,
+      getSize,
+      snapshot,
+      clear,
+    } = collection;
+    function* entries(patt) {
+      for (const k of keys(patt)) {
+        yield [k, k];
+      }
+    }
+    function addAll(elems) {
+      for (const elem of elems) {
+        init(elem, null);
+      }
+    }
+
+    const setStore = {
+      has,
+      add: elem => init(elem, null),
+      addAll,
+      delete: del,
+      keys: patt => keys(patt),
+      values: patt => keys(patt),
+      entries,
+      size,
+      getSize: patt => getSize(patt),
+      snapshot,
+      clear,
+    };
+    return Far('setStore', setStore);
+  }
+
+  function collectionToWeakSetStore(collection) {
+    const { has, init, delete: del } = collection;
+    function addAll(elems) {
+      for (const elem of elems) {
+        init(elem, null);
+      }
+    }
+
+    const weakSetStore = {
+      has,
+      add: elem => init(elem, null),
+      addAll,
+      delete: del,
+    };
+    return Far('weakSetStore', weakSetStore);
+  }
+
+  /**
+   * Produce a *scalar* big map: keys can only be atomic values, primitives, or
+   * remotables.
+   *
+   * @template K,V
+   * @param {string} [label='map'] - diagnostic label for the store
+   * @param {Partial<StoreOptions>=} options
```

Comment:
My fault. See https://github.com/Agoric/agoric-sdk/issues/4412 , including my question for @FUDCo at the end.